Mathlib/Topology/Category/Stonean/Basic.lean

```diff
@@ -0,0 +1,148 @@
+/-
+Copyright (c) 2023 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import Mathlib.Topology.ExtremallyDisconnected
+import Mathlib.CategoryTheory.Sites.Coherent
+import Mathlib.Topology.Category.CompHaus.Projective
+import Mathlib.Topology.Category.Profinite.Basic
+/-!
+# Extremally disconnected sets
+
+This file develops some of the basic theory of extremally disconnected sets.
+
+## Overview
+
+This file defines the type `Stonean` of all extremally (note: not "extremely"!)
+disconnected compact Hausdorff spaces, and gives it the structure of a large category.
+
+The Lean implementation: a term of type `Stonean` is a pair, considering of
+a term of type `CompHaus` (i.e. a compact Hausdorff topological space) plus
+a proof that the space is extremally disconnected.
+This is equivalent to the assertion that the term is projective in `CompHaus`,
+in the sense of category theory (i.e., such that morphisms out of the object
+can be lifted along epimorphisms).
+
+This file defines the type of all Stonean spaces, gives it the
+structure of a large category, and proves some basic observations about this
```

Comment:
This mostly repeats line 17. I think you can remove this paragraph (and maybe add the "basic observations about ..." bit to the paragraph on L17.